Exercise_04_09_new.py

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pandas as pd
+import numpy as np
+import seaborn as sns
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+
+# In[ ]:
+
+
+####1####
+
+
+# In[3]:
+
+
+data=pd.read_csv("C:/Users/kphom4872/Desktop/Training/satday-4-sept-quiz/bike_sharing_hours.csv")
+
+
+# In[4]:
+
+
+data.shape
+
+
+# In[5]:
+
+
+data.head(2)
+
+
+# In[75]:
+
+
+df1=data.loc[data["mnth"] <= 3]
+
+
+# In[76]:
+
+
+df1.head(2)
+
+
+# In[77]:
+
+
+df2=pd.DataFrame(data.groupby(['mnth'])["temp"].mean())
+
+
+# In[78]:
+
+
+df2
+
+
+# In[79]:
+
+
+quat_1=df2[0:3].mean()
+quat_1
+
+
+# In[80]:
+
+
+quat_3=df2[6:9].mean()
+quat_3
+
+
+# In[69]:
+
+
+from scipy.stats import ttest_1samp
+
+
+# In[81]:
+
+
+ttest_1samp(df1['temp'],popmean=quat_1)
+
+
+# In[82]:
+
+
+####Null Hypothesis: The change in average temperature in the first quarter of the year is not significantly different from that in the third quarter of the year 
+####Alternate Hypothesis: The change in average temperature in the first quarter of the year is significantly different from that in the third quarter of the year 
+
+
+# In[83]:
+
+
+###### b) As the p-value is greater than 0.05, therefore we donot reject the null hypothesis and conclude that the change in average temperature in the first quarter of the year is not significantly different from that in the third quarter of the year  
+
+
+# In[ ]:
+
+
+####### c)
+
+
+# In[84]:
+
+
+data.head(2)
+
+
+# In[97]:
+
+
+weekday_grps=[data.loc[data['weekday']==q,'casual'].values 
+              
+              for q in data['weekday'].unique() ]
+
+
+# In[98]:
+
+
+from scipy.stats import f_oneway
+
+
+# In[99]:
+
+
+f_oneway(*weekday_grps)
+
+
+# In[ ]:
+
+
+##### As the p-valye is less than 0.05, therefore we reject the null hypothesis and conclude that the casually rented bikes are dependent on the weekdays.
+
+
+# In[ ]:
+
+
+
+
+
+# In[7]:
+
+
+pd.options.display.max_columns=None
+pd.options.display.max_rows=None
+
+
+# In[37]:
+
+
+data["mnth"].value_counts(dropna=False)
+
+
+# In[9]:
+
+
+data["cnt"].max()
+
+
+# In[ ]:
+
+
+################## Q2 #######################
+
+
+# In[ ]:
+
+
+
+
+
+# In[ ]:
+
+
+
+
+
+# In[ ]:
+
+
+
+
+
+# In[ ]:
+
+
+
+
+
+# In[ ]:
+
+
+
+
+
+# In[ ]:
+
+
+
+
+
+# In[ ]:
+
+
+
+
+
+# In[ ]:
+
+
+############## Q3 #############
+
+
+# In[10]:
+
+
+list(zip(data.columns,data.nunique(),data.dtypes))
+
+
+# In[13]:
+
+
+sns.jointplot(x="temp", y="cnt", data=data)
+
+
+# In[24]:
+
+
+sns.lmplot('temp', 'cnt', 
+           data=data.iloc[1:100,:],palette="Set1",
+           fit_reg=True,order=2)
+
+
+# In[ ]:
+
+
+#### As the graph is quite random, so there is no correlation between temp and bike renting behaviours. 
+
+
+# In[25]:
+
+
+sns.jointplot(x="hum", y="cnt", data=data)
+
+
+# In[100]:
+
+
+sns.lmplot('hum', 'cnt', 
+           data=data.iloc[1:100,:],palette="Set1",
+           fit_reg=True,order=2)
+
```

Comment:
Also you dont always have to subset data, in fact you shouldn't most of the time .

Reply:
Yaa, got this point.